src-ghc/Pact/Server/ApiServer.hs

```diff
@@ -156,14 +155,21 @@ buildCmdRpc c@Command {..} = do
   log $ "Processing command with hash: " ++ show _cmdHash
   return (cmdToRequestKey c,fmap encodeUtf8 c)
 
-group :: Int -> [a] -> [[a]]
-group _ [] = []
+-- | Reorganize a `NonEmpty` into sub-batches.
+group :: Int -> NonEmpty a -> NonEmpty (NonEmpty a)
 group n l
-  | n > 0 = take n l : group n (drop n l)
-  | otherwise = error "Negative n"
+  | n < 1 = error "Non-positive n"
```

Comment:
We should not be using `error` to die like this. This should probably use the `die' "Empty Batch"` error previously deleted above

Reply:
Luckily this function is only used once and called once (in this module), so the chance of this error firing is low. The real solution is `Positive Word` via `refined` :)

Reply:
My real objection to this is that a simple conversion to and from list would have worked fine, instead this is both hard to understand and unsafe.